src/business_logic/transaction/fee.rs

```diff
@@ -0,0 +1,133 @@
+use super::error::TransactionError;
+use crate::{
+    business_logic::{
+        execution::{
+            execution_entry_point::ExecutionEntryPoint,
+            gas_usage,
+            objects::{CallInfo, TransactionExecutionContext},
+        },
+        fact_state::{
+            in_memory_state_reader::InMemoryStateReader, state::ExecutionResourcesManager,
+        },
+        state::cached_state::CachedState,
+    },
+    definitions::general_config::{self, StarknetGeneralConfig},
+    services::api::contract_class::EntryPointType,
+};
+use felt::Felt;
+use num_traits::{Num, ToPrimitive};
+use std::cmp;
+use std::collections::{HashMap, HashSet};
+
+// ----------------------------------------------------------------------------------------
+/// Transfers the amount actual_fee from the caller account to the sequencer.
+/// Returns the resulting CallInfo of the transfer call.
+// ----------------------------------------------------------------------------------------
```

Comment:
I think those kind of comments can be confusing since I'd expect them to be kind of like a section separator, while in this case it's only the function's documentation.

Reply:
I agree with this. Let's use standard Rust doc comments for function comments.

Reply:
Sense of beauty destroyed

Reply:
Almost. I'd remove the blank line between the comment and the actual `fn documented_function()` so that it becomes evident at first glance that it's that function's documentation and not random text.

Reply:
makes sense